src/Extensions/Diagnostics.HealthChecks/Internal/HealthCheckPublisher.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Fabric;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.ObjectPool;
+using ServiceFabricHealth = System.Fabric.Health;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
+{
+	internal abstract class HealthCheckPublisher : IHealthCheckPublisher
+	{
+		protected readonly ObjectPool<StringBuilder> StringBuilderPool;
+
+		internal static string HealthReportSourceId => nameof(HealthCheckPublisher);
+
+		internal const string HealthReportSummaryProperty = "HealthReportSummary";
+
+		public HealthCheckPublisher(ObjectPoolProvider objectPoolProvider) => StringBuilderPool = objectPoolProvider.CreateStringBuilderPool();
+
+		public abstract Task PublishAsync(HealthReport report, CancellationToken cancellationToken);
+
+		protected abstract ServiceFabricHealth.HealthInformation FinalizeHealthReport(string healthReportSummaryProperty, ServiceFabricHealth.HealthState healthState);
```

Comment:
Also for my information, could you please explain what's the point of making this method abstract, it looks identical in both cases?

Reply:
Yes, sure - it is only for taking proper Publisher Source Id  because static members can't be abstract

Reply:
In this case, I think just making Publisher Source Id property non-static and abstract would be much better, than splitting this method.